### DISCUSSION TITLE
add meshopt decompression to copy cli

### DISCUSSION THREAD

**Discussion by arpu** [donmccurdy/glTF-Transform#582] [General]
the cli copy command decompress draco compressed Assets automatic at the moment but not mesh_opt 

 what i read gltf-transform use the inbuild transform()  https://github.com/donmccurdy/glTF-Transform/blob/main/packages/cli/src/cli.ts#L142

is there a example script to do this with meshopt registered extention?

**Comment by donmccurdy** [comment]
Hmm it's definitely supposed to, and seems to work for me:

https://github.com/donmccurdy/glTF-Transform/blob/6671e0c480ee990bfa119ea946e328bbe91f4233/packages/cli/src/session.ts#L24-L34

Do you have an example model that doesn't decompress as expected?

**Comment by arpu** [comment]
you are right, not sure why this not worked on the server need some more investigation

**Comment by arpu** [comment]
ok, the problem comes from the basisu compressed images, would it be possible to unpack x.basis to PNG  on the copy command?

**Comment by donmccurdy** [comment]
The `copy` command only removes Draco compression because it's necessary when unpacking the file to an in-memory [Document](https://gltf-transform.donmccurdy.com/classes/core.document.html). Decompressing .ktx2 textures isn't required, and so I wouldn't want the unnecessary loss from unpacking .ktx2 files by default — they'll lose quality if recompressed.

I do think a separate command (`decompress`?) would be a good feature request. I would probably want to wait to implement that myself until the [Khronos lightweight transcoders](https://github.com/KhronosGroup/Basis-Universal-Transcoders) support ETC1S -> RGBA transcoding first.
